### PR TITLE
Enable website to load dynamic scripts from manifest JS file

### PR DESCRIPTION
**Pull request by Jahnp**
#### Pull request checklist

- [ ] Addresses an existing issue: N/A (enables better site deployment management)
- [x] Include a change request file using `$ npm run change`

#### Description of changes
Today, the website loads all of its app scripts from a single, static entry point file. Pushing updates to the site means that we need to continuously update that single entry point and add 70+ bundle chunks. 

The changes in this PR enable us to load the website entry point and chunks from a dynamic location, determined by a simple manifest JS file. It is one part of a set of infrastructure updates we're undertaking to allow the site to be updated with every release of Fabric, and roll back faulty changes quickly and predictably.

### CODE REVIEW

File: apps/fabric-website/index.html [hunk 1]
```diff
@@ -29,32 +28,107 @@
 <body>
   <div id="main">
     <div class="loading">
-      <img class="loadingImage" src="https://static2.sharepointonline.com/files/fabric/fabric-website/images/spinner.gif" alt="Loading"
-        width="32" height="32" />
+      <img class="loadingImage" src="https://static2.sharepointonline.com/files/fabric/fabric-website/images/spinner.gif"
```
Comment (Jahnp) [inline apps/fabric-website/index.html]: Prettier change

File: apps/fabric-website/src/root.tsx [hunk 2]
```diff
@@ -25,10 +25,19 @@ initializeIcons();
 
 let isProduction = process.argv.indexOf('--production') > -1;
 
+let isLocal = window.location.hostname === 'localhost' || window.location.hostname.indexOf('ngrok.io') > -1;
```
Comment (cliffkoh) [inline apps/fabric-website/src/root.tsx]: nice

Comment (mikewheaton) [reply apps/fabric-website/src/root.tsx]: I'd never heard of ngrok before. Signing up now with hopes that it'll make it easier to test across virtual machines without all the shared network issues.

File: apps/fabric-website/index.html [hunk 3]
```diff
@@ -29,32 +28,107 @@
...
+      var reactPath = '//cdnjs.cloudflare.com/ajax/libs/react/16.3.2/umd/react';
+      var reactDomPath = '//cdnjs.cloudflare.com/ajax/libs/react-dom/16.3.2/umd/react-dom';
+
+      if (noCompress) {
+        reactPath = reactPath + '.development.js';
+        reactDomPath = reactDomPath + '.development.js'
+        appPath += entryPointFilename + '.js';
+      } else {
+        reactPath = reactPath + '.production.min.js';
+        reactDomPath = reactDomPath + '.production.min.js'
```
Comment (KevinTCoughlin) [inline apps/fabric-website/index.html]: Non-blocking nit, but perhaps add the missing semi-colon here for consistency

File: apps/fabric-website/index.html [hunk 4]
```diff
@@ -29,32 +28,107 @@
...
-    var isLocal = window.location.hostname === 'localhost';
-    // This path allows minified files to be served locally for testing purposes.
-    var appPath = isLocal ? '' : '';
+    function loadAppScripts(appPath) {
+      var reactPath = '//cdnjs.cloudflare.com/ajax/libs/react/16.3.2/umd/react';
+      var reactDomPath = '//cdnjs.cloudflare.com/ajax/libs/react-dom/16.3.2/umd/react-dom';
+
+      if (noCompress) {
+        reactPath = reactPath + '.development.js';
+        reactDomPath = reactDomPath + '.development.js'
```
Comment (KevinTCoughlin) [inline apps/fabric-website/index.html]: here too

File: apps/fabric-website/src/root.tsx (hunk 2 above)
Comment (KevinTCoughlin) [reply apps/fabric-website/src/root.tsx]: Its pretty awesome

File: apps/fabric-website/index.html [hunk 5]
```diff
@@ -29,32 +28,107 @@
...
+        }
+      })();
     }
 
-    var isLocal = window.location.hostname === 'localhost';
-    // This path allows minified files to be served locally for testing purposes.
-    var appPath = isLocal ? '' : '';
+    function loadAppScripts(appPath) {
+      var reactPath = '//cdnjs.cloudflare.com/ajax/libs/react/16.3.2/umd/react';
+      var reactDomPath = '//cdnjs.cloudflare.com/ajax/libs/react-dom/16.3.2/umd/react-dom';
```
Comment (KevinTCoughlin) [inline apps/fabric-website/index.html]: Just noting that we're bumping from `16.3.1 => 16.3.2` with this change

File: apps/fabric-website/index.html [hunk 6]
```diff
@@ -29,32 +28,107 @@
 <body>
   <div id="main">
     <div class="loading">
-      <img class="loadingImage" src="https://static2.sharepointonline.com/files/fabric/fabric-website/images/spinner.gif" alt="Loading"
-        width="32" height="32" />
+      <img class="loadingImage" src="https://static2.sharepointonline.com/files/fabric/fabric-website/images/spinner.gif"
+        alt="Loading" width="32" height="32" />
```
Comment (KevinTCoughlin) [inline apps/fabric-website/index.html]: 👍

**Comment by KevinTCoughlin** [comment]
>LGTM. Only feedback maybe is that I'm a tad concerned about the heft of this bootstrapping code in production - I don't think it will be minified in the pipeline, right?

I think you're right. We'd have to move the scripts to be part of our webpack build as a separate output and inject it for it to be minified.

File: apps/fabric-website/index.html (hunk 5 above)
Comment (Jahnp) [reply apps/fabric-website/index.html]: Good callout. One thing to note here is that the public, hosted site currently doesn't use this method to load React--they're baked into the page by the server. I'd like to change that soon so we can completely self-manage this through Fabric. But in the meantime, it means that this piece will only be relevant for a.) local testing, and b.) the staging site I'm setting up to act as a dogfood environment.

File: apps/fabric-website/index.html (hunk 3 above)
Comment (Jahnp) [reply apps/fabric-website/index.html]: Oh goodness, how embarrassing. I'll fix this--thanks!

**Comment by Jahnp** [comment]
@cliffkoh @KevinTCoughlin you're absolutely right on this point. Really, much of this functionality should be handled server-side, but due to current infra limitations we can't do much there at the moment. Updating the actual HTML page is a manual process, so this mechanism is meant to be a workaround.

In the short term, we *could* consider moving this logic to a separate bundle, but then we'd introduce another request to fetch that file. The best thing really is to work towards adding more server-side configurability--while we work towards that, are we fine moving ahead with this approach?

**Pull request merged**